packages/mongo/remote_collection_driver.ts

```diff
@@ -121,10 +121,26 @@ MongoInternals.defaultRemoteCollectionDriver = once((): RemoteCollectionDriver =
 
   // Initialize database connection on startup
   Meteor.startup(async (): Promise<void> => {
-    await driver.mongo.client.connect();
+    const isDevelopment = Meteor.isDevelopment;
+    return new Promise((resolve, reject) => {
+      driver.mongo.client
+        .connect()
+        .then(resolve)
+        .catch((error) => {
+          // Fix an issue with development mode crashing on idle
+          // https://github.com/meteor/meteor/issues/13108
+          if (isDevelopment &&
+            (error?.message?.includes("PoolClearedOnNetworkError") ||
+              error?.message?.includes("server monitor timeout"))
+          ) {
+            return;
+          }
+          reject(error);
+        });
+    });
```

Comment:
maybe a retry on these cases? 

```suggestion
  const connectToRemoteDatabase = async (retries = 3) => {
    if (retries === 0) throw new Error("Failed to connect to remote database");

    try {
      await driver.mongo.client.connect();
    } catch (error) {
      if (
        Meteor.isDevelopment &&
        (error?.message?.includes("PoolClearedOnNetworkError") ||
          error?.message?.includes("server monitor timeout"))
      ) {
        return await connectToRemoteDatabase(retries - 1);
      }
      throw error;
    }
  };

  // Initialize database connection on startup
  Meteor.startup(async (): Promise<void> => {
    return await connectToRemoteDatabase();
  });
```

Reply:
So far no issue, but I no longer see the `connectionPoolCleared` log anymore. Not sure why. I also added other error checking.

Continuing testing, if I get good state on my last checking, maybe is good for members of the community to test them and see how this fix behaves to them.